src/main/java/uk/gov/ons/census/fwmtadapter/client/CaseClient.java

```diff
@@ -20,17 +20,14 @@ public CaseClient(RestTemplate restTemplate) {
     this.restTemplate = restTemplate;
   }
 
-  public String getCaseIdFromQid(String questionnaire_id) {
-    String url = "http://" + host + ":" + port + "/cases/qid/" + questionnaire_id;
-
-    CaseIdDto caseIdDto = restTemplate.getForObject(url, CaseIdDto.class);
-
-    String caseId = caseIdDto.getCaseId();
-
-    if (StringUtils.isEmpty(caseId)) {
-      throw new RuntimeException("Returned empty caseID from case api");
-    }
+  public CaseContainer getCaseFromCaseId(String caseId) {
+    String url = "http://" + host + ":" + port + "/cases/" + caseId;
```

Comment:
Prefer `String.format`

Reply:
also, this is how injection attacks happen. It's better to use `UriComponentsBuilder`